src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -0,0 +1,681 @@
+
+        public abstract long[] columnSumsToArray(final long[] dest);
+
+        public int columnCount() {
+            return columnNames().size();
+        }
+    }
+
+    /**
+     * Counts class to be use when {@link #groupBy} is {@code GroupBy#COHORT}.
+     */
+    private class TotalCounts extends Counts {
+
+        final int[] counts;
+
+        private final String columnName;
+
+        private long columnSum;
+
+        /**
+         * Creates new cohort (all-in-one-count) counts collection.
+         *
+         * @throws IllegalStateException if {@link #exonDB} or {@link #sampleDB} have not been
+         *   properly initialized.
+         */
+        public TotalCounts() {
+            counts = new int[exonDB.exonCount()];
+            columnName = calculateColumnName(cohortName,countColumnName);
+            columnSum = 0;
+        }
+
+        // Make substitutions in the column name given the default cohort-name, count-column-format and the
+        // sample database.
+        private String calculateColumnName(final String cohortName, final String countColumnName) {
+            if (countColumnName.equals("%^"))
+                return cohortName;
+            String result = countColumnName;
+            result = result.replaceAll("%c", cohortName).replaceAll("%\\^",cohortName);
+
+            if (result.contains("%s")) {
+                if (sampleDB.sampleCount() > 1)
+                    throw new UserException.BadArgumentValue("countColumnName",countColumnName,
+                            " cannot use %s (sample-id) in column count header with group-by COHORT (default) and multiple samples");
+                final String sampleId = sampleDB.sampleCount() == 1 ? sampleDB.sampleIds().get(0) : ANONYMOUS_SAMPLE_ID;
+                result = result.replaceAll("%s",sampleId);
+            }
+            if (result.contains("%g")) {
+                if (sampleDB.readGroupCount() > 1)
+                    throw new UserException.BadArgumentValue("countColumnName",countColumnName,
+                            " cannot use %g (read-group-id) in column count header grouping all counts (default) and with multiple read-groups in the inputs");
+                final String groupId = sampleDB.readGroupCount() == 1 ? sampleDB.readGroupIds().get(0) : ANONYMOUS_READ_GROUP_ID;
+                result = result.replaceAll("%g",groupId);
+            }
+            return result;
+        }
+
+        @Override
+        public void count(final SAMRecord read, final int exonIndex) {
+
+            counts[exonIndex]++;
+            columnSum++;
+        }
+
+        @Override
+        public List<String> columnNames() {
+            return Collections.singletonList(columnName);
+        }
+
+        @Override
+        public void copyExonValues(final int exonIndex, final int[] dest) {
+            dest[0] = counts[exonIndex];
+        }
+
+        @Override
+        public long[] columnSumsToArray(final long[] dest) {
```

Comment:
`Collection.toArray` has no other choice because of deficient type system in Java (you can't call `new T[size]` so they need to take in an array). The consideration here is completely different. And the harm is very real - the human reader is tricked by the unexpected side effect of this call.